mono/utils/mono-os-mutex.h

```diff
@@ -59,7 +59,7 @@ mono_os_mutex_init_type (mono_mutex_t *mutex, int type)
 	if (G_UNLIKELY (res != 0))
 		g_error ("%s: pthread_mutexattr_settype failed with \"%s\" (%d)", __func__, g_strerror (res), res);
 
-#if !defined(__HAIKU__) && defined (PTHREAD_PRIO_INHERIT) && HAVE_DECL_PTHREAD_MUTEXATTR_SETPROTOCOL
+#if !defined(__HAIKU__) && !defined(MUSL) && defined (PTHREAD_PRIO_INHERIT) && HAVE_DECL_PTHREAD_MUTEXATTR_SETPROTOCOL
 	/* use PTHREAD_PRIO_INHERIT if possible */
```

Comment:
Usually instead of testing for musl testing for glibc is done, so maybe `&& defined(__GLIBC)` does the trick?

Reply:
That asssumes other libc has the same bug as muslc.
It is a bug in muslc, right?

Reply:
Not sure about that, I just think that detecting this via the triplet might be a bit fragile. I suppose the musl folks would know more, maybe you could contact them in #musl on freenode. Anyway, thanks for looking into this :)

Reply:
> `defined(__GLIBC__)`

I considered that first, but I took a more conservative approach since I have no way of testing other libc implementations - I only know this fixes the issue on muslc.

I am not sure about other OSes that potentially uses pthread but not glibc? I can only find the cases Pthread, Win32 and no threads. I wouldn't know if my fix breaks on e.g. Solaris.

That said, I am happy to change it if there is a better solution.

Reply:
We should prefer the `!defined(MUSL)` fix here because multiple OSes may hit `defined(__GLIBC__)` even when unintended. This isn't necessarily a bug as much as a difference in how pthreads and timers behave on different implementations. FreeBSD exhibits basically identical behavior, just doesn't go fully broken.

We should probably also add a 'todo' to make autoconf more `musl`-aware as a whole than it is currently.

Reply:
Ah okie :+1: